src/Microsoft.Azure.WebJobs.Extensions.OpenApi.Core/SwaggerUI.cs

```diff
@@ -88,9 +84,7 @@ public ISwaggerUI AddServer(IHttpRequestDataObject req, string routePrefix, IOpe
             }
 
             this._baseUrl = servers.First().Url;
```

Comment:
If someone sets the server URL like `/api`, this `_baseUrl` value will be in trouble at the following `Uri(...)` instantiation. Therefore, we need additional logic to check whether the given `servers.First().Url` starts with fully-qualified domain name or not. If it doesn't start with FQDN, we should add the logic something like:

```csharp
if (!this._baseUrl.StartsWith("http"))
{
    this._baseUrl = $"{this._req.GetScheme(options)}://{this._req.Host}{this._baseUrl}";
}
```

Reply:
Adding this logic can still be problematic in some cases. For example, we have Azure Functions `func-1` and `func-2`, and an Azure Front Door `funcfd`.
We can't use Custom Domains on Azure Functions because the front door domain not controlled by customers and not unique across instances.
When user accessing our service through `funcfd.azurefd.net`, the actual backend host `func-1.azurewebsites.net` or `func-2.azurewebsites.net` will be passed to backend Azure Functions. Thus the base URL returned will be `https://func-1.azurewebsites.net/api` or `https://func-2.azurewebsites.net/api` with the above logic, and resulting in cross origin requests on Swagger UI from Front Door domain.

Reply:
That makes sense.